File: php/hamming/hamming.php
<?php

/*
This is only a SKELETON file for the "Hamming" exercise. It's been provided as a
convenience to get you started writing code faster.

Remove this comment before submitting your exercise.
*/

function distance(string $strandA, string $strandB) : int
{
    if (strlen($strandA) != strlen($strandB)) {
        throw new InvalidArgumentException("DNA strands must be of equal length.");
    }

    return count(array_diff_assoc(str_split($strandA), str_split($strandB)));
}

File: ruby/hamming/hamming.rb
module Hamming
  def self.compute(strand_a, strand_b)
    raise ArgumentError, 'strand length mismatch' unless strand_a.length == strand_b.length

    strand_a.chars.zip(strand_b.chars).count { |l1, l2| l1 != l2 }
  end
end
